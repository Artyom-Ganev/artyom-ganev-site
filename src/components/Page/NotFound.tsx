import React from 'react';
import { Link } from 'react-router-dom';
import Container from './Container';
import './NotFound.scss';

const TITLE = 'Page Not Found';

/**
 * 404 page not component
 */
const NotFound = () => (
  <Container title={TITLE}>
    <div className='page-notFound__item'>
      <Link to='/'>Go to main page</Link>
    </div>
  </Container>
);

export default NotFound;
