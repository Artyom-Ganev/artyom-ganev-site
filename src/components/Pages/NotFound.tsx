import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import { PageContainer } from 'src/components/Pages';
import styles from './NotFound.module.scss';

const TITLE = 'Page Not Found';

/**
 * 404 page not component
 */
const NotFound: FunctionComponent = () => (
  <PageContainer title={TITLE}>
    <div className={styles.container}>
      <Link to='/'>Go to main page</Link>
    </div>
  </PageContainer>
);

export default NotFound;
