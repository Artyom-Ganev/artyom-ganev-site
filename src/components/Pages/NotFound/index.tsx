import Typography from '@material-ui/core/Typography';
import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import { PageContainer } from 'src/components/Pages';

const TITLE = 'Page Not Found';

/**
 * 404 page not component
 */
const NotFound: FunctionComponent = () => {
  return (
    <PageContainer title={TITLE}>
      <Typography paragraph variant='h4'>
        <Link to='/'>Go to main page</Link>
      </Typography>
    </PageContainer>
  );
};

export default NotFound;
