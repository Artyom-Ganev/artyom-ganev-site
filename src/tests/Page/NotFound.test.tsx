import React from 'react';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NotFound from '../../components/Page/NotFound';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';

test('NotFound', () => {
  const app = () => ({ loading: false });

  const store = createStore(combineReducers({ app }));

  const component = renderer.create(
    <Provider store={store}>
      <HashRouter>
        <NotFound />
      </HashRouter>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
