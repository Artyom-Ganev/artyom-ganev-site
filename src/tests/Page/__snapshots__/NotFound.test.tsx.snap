// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound 1`] = `
<div
  className="flexBox flexColumn page-container__root"
>
  <h2>
    Page Not Found
  </h2>
  <div
    className="page-notFound__item"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      Go to main page
    </a>
  </div>
</div>
`;
